src/components/DatePickerModal/index.js

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Header, HeaderTitle, StyledModal, StyledCalendar } from './styled';
+
+export default function DatePickerModal(props) {
+    const {
+        id,
+        className,
+        style,
+        locale,
+        minDate,
+        maxDate,
+        selectionType,
+        variant,
+        value,
+        onChange,
+        onRequestClose,
+        isOpen,
+        title,
+    } = props;
+
+    return (
+        <StyledModal
+            id={id}
+            className={className}
+            style={style}
+            isOpen={isOpen}
+            onRequestClose={onRequestClose}
+        >
+            <Header>
+                <HeaderTitle>{title}</HeaderTitle>
+            </Header>
+            <StyledCalendar
+                locale={locale}
+                minDate={minDate}
+                maxDate={maxDate}
+                selectionType={selectionType}
+                variant={variant}
+                value={value}
+                onChange={onChange}
+            />
+        </StyledModal>
+    );
+}
+
+DatePickerModal.propTypes = {
+    /** The id of the outer element. */
+    id: PropTypes.string,
+    /** A CSS class for the outer element, in addition to the component's base classes. */
+    className: PropTypes.string,
+    /** An object with custom style applied to the outer element. */
+    style: PropTypes.object,
+    /**
+     * The DatePickerModal locale. Defaults to browser's language.
+     */
+    locale: PropTypes.string,
+    /** The ending of a range of valid dates. The range includes the endDate.
+     * The default value is current date + 100 years. */
+    maxDate: PropTypes.instanceOf(Date),
+    /** The beginning of a range of valid dates. The range includes the startDate.
+     * The default value is current date - 100 years. */
+    minDate: PropTypes.instanceOf(Date),
+    /** The type of the selection. It can be a single date or a range. The default value is 'single'. */
+    selectionType: PropTypes.oneOf(['single', 'range']),
+    /** The calendar type. Defaults to 'single' */
+    variant: PropTypes.oneOf(['single', 'double']),
+    /** The action triggered when a value attribute changes. */
+    onChange: PropTypes.func,
+    /** Sets the date for the DatePickerModal programmatically. */
+    value: PropTypes.oneOfType([
+        PropTypes.instanceOf(Date),
+        PropTypes.string,
+        PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.instanceOf(Date), PropTypes.string])),
+    ]),
+    /** Controls whether the DatePickerModal is opened or not. */
+    isOpen: PropTypes.bool,
+    /** The action triggered when the component request to close
+     *  (e.g click close button, press esc key or click outside). */
+    onRequestClose: PropTypes.func,
+    /** The title can include text or another component,
+     * and is displayed in the header of the component. */
+    title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
```

Comment:
The title in the modal will be shown according to the selected date, which is the use case we have now. 
We will not have this prop for now if in the future the use case appears where you need to pass it a customizable title then we will add the prop.